### PR TITLE
Added logging to Gitrise and also cleaned the code.

### PR DESCRIPTION
**Pull request by Azbagheri**

### CODE REVIEW

File: gitrise.sh [hunk 1]
```diff
@@ -61,18 +62,21 @@ while [ $# -gt 0 ]; do
...
-        POSITIONAL+=("$1")
         exit 1
     ;;
     esac
 done
 
-# restore positional parameters
-set -- "${POSITIONAL[@]}"
+# Create temp directory if debugging mode enabled
+[ "$DEBUG" == "true" ] && mkdir -p gitrise_temp
```
Comment (aroswell) [inline gitrise.sh]: If a gitrise_temp directory exist should we delete it every time the trigger runs.

File: tests/gitrise_logging_tests.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#! /bin/bash 
+# shellcheck disable=SC1091,SC2155,SC2154
+# Not following: (error message here)
+
+source ./gitrise.sh -t -d
+
+testDebugMode(){
+    assertTrue "TDEBUG is not set to 'true'" "$DEBUG"
```
Comment (aroswell) [inline tests/gitrise_logging_tests.sh]: typo perhaps? `TDEBUG`

Comment (Azbagheri) [reply tests/gitrise_logging_tests.sh]: yeah, good catch! :)

File: gitrise.sh (hunk 1 above)
Comment (Azbagheri) [reply gitrise.sh]: that's a good point. Changed the if statement to check that first.

**Pull request merged**